safe/gui/widgets/dock.py

```diff
@@ -1478,8 +1479,20 @@ def restore_function_state(self, original_function):
                 break
 
     def print_map(self):
-        """Slot to open impact report dialog that used to tune report
-        when print map button pressed."""
+        """Open impact report dialog that used to tune report when print map
+            button pressed."""
+        # Check if selected layer is valid
+        impact_layer = self.iface.activeLayer()
+        if impact_layer is None:
```

Comment:
Maybe we should also check the keywords here to make sure it really is an impact summary layer.